nibabel/streamlines/tests/test_tractogram.py

```diff
@@ -688,6 +688,22 @@ def test_tractogram_apply_affine(self):
                            np.dot(np.eye(4), np.dot(np.linalg.inv(affine),
                                                     np.linalg.inv(affine))))
 
+        # Applying the affine to a tractogram that has been indexed or sliced
+        # shouldn't affect the remaining streamlines.
+        tractogram = DATA['tractogram'].copy()
+        transformed_tractogram = tractogram[::2].apply_affine(affine)
+        assert_true(transformed_tractogram is not tractogram)
+        check_tractogram(tractogram[::2],
+                         streamlines=[s*scaling for s in DATA['streamlines'][::2]],
+                         data_per_streamline=DATA['tractogram'].data_per_streamline[::2],
+                         data_per_point=DATA['tractogram'].data_per_point[::2])
```

Comment:
This feels very strange, but to be clear, is the following the intended semantics?

```Python
sliced_tractogram = tractogram[::2]  # A "view" tractogram
transformed_tractogram = sliced_tractogram.apply_affine(affine)
assert transformed_tractogram is sliced_tractogram
assert np.array_equal(tractogram[::2].data, sliced_tractogram.data)
```

Reply:
Kinda but `Tractogram` doesn't have a `.data` property. `LazyTractogram` does have one but it returns a generator over TractogramItem, not a ndarray. In the test, I'm also looking at `data_per_point` which is a dictionary of `ArraySequence`s.